File: java/src/test/java/jooqsimple/model/PersonRepositoryTest.java
package jooqsimple.model;

import org.junit.jupiter.api.*;

import java.util.*;

import static org.junit.jupiter.api.Assertions.*;

class PersonRepositoryTest extends AbstractIntegrationTest {

    @BeforeEach
    public void reset() {
        new PersonRepository(dslContext).deleteAll();
    }

    @Test
    void insertAndSearch() {
        // Given
        PersonRepository personRepo = new PersonRepository(dslContext);

        personRepo.insertPerson(
                new Person(
                        1L,
                        "First",
                        "Person",
                        List.of(new Address("Line1", "Line2"))));
        personRepo.insertPerson(
                new Person(
                        2L,
                        "Second",
                        "Person",
                        new ArrayList<>()));

        // When
        var retrievedPersons = personRepo.search("First", "Person");

        // Then
        assertEquals(1, retrievedPersons.size());
        assertEquals(1, retrievedPersons.get(0).addresses().size());
    }

}

File: kotlin/src/test/kotlin/jooqsimple/model/PersonRepositoryTest.kt
package jooqsimple.model;

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class PersonRepositoryTest : AbstractIntegrationTest() {

    @BeforeEach
    fun reset() {
        val personRepo = PersonRepository(dslContext)
        personRepo.deleteAll()
    }

    @Test
    fun insertAndSearch() {
        val personRepo = PersonRepository(dslContext)
        // Given
        personRepo.insertPerson(
            Person(
                1L,
                "First",
                "Person",
                listOf(Address("Line1", "Line2"))
            )
        )
        personRepo.insertPerson(
            Person(
                2L,
                "Second",
                "Person",
                listOf()
            )
        )

        // When
        val retrievedPersons = personRepo.search("First", "Person");

        // Then
        assertEquals(1, retrievedPersons.size);
        assertEquals(1, retrievedPersons[0].addresses.size);
    }

}
